packages/governance/src/governor/governor.cairo

```diff
@@ -0,0 +1,1042 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.19.0 (governance/governor/governor.cairo)
+
+/// # Governor Component
+///
+/// Core of the governance system.
+#[starknet::component]
+pub mod GovernorComponent {
+    use core::hash::{HashStateTrait, HashStateExTrait};
+    use core::num::traits::Zero;
+    use core::pedersen::PedersenTrait;
+    use crate::governor::ProposalCore;
+    use crate::governor::interface::{ProposalState, IGovernor, IGOVERNOR_ID};
+    use crate::governor::vote::{Vote, VoteWithReasonAndParams};
+    use crate::utils::HashSpanImpl;
+    use crate::utils::call_impls::HashCallImpl;
+    use openzeppelin_account::interface::{ISRC6Dispatcher, ISRC6DispatcherTrait};
+    use openzeppelin_introspection::src5::SRC5Component::InternalImpl as SRC5InternalImpl;
+    use openzeppelin_introspection::src5::SRC5Component;
+    use openzeppelin_utils::bytearray::ByteArrayExtTrait;
+    use openzeppelin_utils::cryptography::snip12::{OffchainMessageHash, SNIP12Metadata};
+    use starknet::account::Call;
+    use starknet::storage::{Map, StorageMapReadAccess, StorageMapWriteAccess};
+    use starknet::{ContractAddress, SyscallResultTrait};
+
+    type ProposalId = felt252;
+
+    #[storage]
+    pub struct Storage {
+        pub Governor_proposals: Map<ProposalId, ProposalCore>,
+        pub Governor_nonces: Map<ContractAddress, felt252>
```

Comment:
Because we need to add an extra component dependency both in the Governor component and later in the preset, and since there is a lot already, I'm trying to save as many required embeddings and dependencies as possible.

Reply:
I agree it is verbose...well, two things:
1. I really don't love the idea. It's only two fns and a mapping, so it's not the worst thing...but we are repeating ourselves and increasing the code of an already complex component
2. To simplify deps a bit, WDYT about moving governor_core_execution into the component itself? It has neither events nor storage and contracts would have one less `component!` to declare

Reply:
All that said, I'm okay with including nonces like this if you want to keep it. We can change it later if/when there's a better pattern or configuration we can use

Reply:
> I really don't love the idea. It's only two fns and a mapping, so it's not the worst thing...but we are repeating ourselves and increasing the code of an already complex component

Between the two options I vote we save the extra embedding for the user. It is a small addition for us, and a significant save for users. User UX has a bigger weight, and we should prioritize it as much as possible.

Reply:
> To simplify deps a bit, WDYT about moving governor_core_execution into the component itself? It has neither events nor storage and contracts would have one less component! to declare

Note we cannot define the implementation inside the component, since then users won't be able to use a different implementation, since they would conflict. We can remove the component boilerplate and make it a module with the implementation, I used a component just to make the extensions consistent and easier to follow for users.

Reply:
> Note we cannot define the implementation inside the component, since then users won't be able to use a different implementation, since they would conflict.

Mmmmm my bad, that's right

> We can remove the component boilerplate and make it a module with the implementation, I used a component just to make the extensions consistent and easier to follow for users.

I was just looking for ways we could consolidate. The current design seems like the correct choice 👍 it'd be weird to not make core execution a component IMO